src/source/Ice/IceAgent.c

```diff
@@ -81,14 +81,25 @@ STATUS createIceAgent(PCHAR username, PCHAR password, PIceAgentCallbacks pIceAge
     CHK_STATUS(createStunPacket(STUN_PACKET_TYPE_BINDING_INDICATION,
                                 NULL,
                                 &pIceAgent->pBindingIndication));
-
+    CHK_STATUS(hashTableCreateWithParams(ICE_HASH_TABLE_BUCKET_COUNT, ICE_HASH_TABLE_BUCKET_LENGTH, &pIceAgent->requestTimestampDiagnostics));
     pIceAgent->iceServersCount = 0;
     for (i = 0; i < MAX_ICE_SERVERS_COUNT; i++) {
         if (pRtcConfiguration->iceServers[i].urls[0] != '\0' &&
             STATUS_SUCCEEDED(parseIceServer(&pIceAgent->iceServers[pIceAgent->iceServersCount],
                                             (PCHAR) pRtcConfiguration->iceServers[i].urls,
                                             (PCHAR) pRtcConfiguration->iceServers[i].username,
                                             (PCHAR) pRtcConfiguration->iceServers[i].credential))) {
+            pIceAgent->rtcIceServerDiagnostics[i].port = (INT32)getInt16(pIceAgent->iceServers[i].ipAddress.port);
+            if(pIceAgent->iceServers[pIceAgent->iceServersCount].transport == KVS_SOCKET_PROTOCOL_UDP) {
```

Comment:
switch case would be more readable here.

Reply:
Hmm. Fair. Fixed.